app/presenters/order_detail_notice_presenter.rb

```diff
@@ -0,0 +1,85 @@
+class OrderDetailNoticePresenter < DelegateClass(OrderDetail)
+
+  include ActionView::Helpers::TagHelper
+  include ActionView::Helpers::OutputSafetyHelper
+
+  def statuses
+    [].tap do |statuses|
+      statuses << Notice.new(:in_review) if in_review?
+      statuses << Notice.new(:in_dispute) if in_dispute?
+      statuses << Notice.new(:can_reconcile) if can_reconcile_journaled?
+      statuses << Notice.new(:in_open_journal) if in_open_journal?
+    end
+  end
+
```

Comment:
I think that I find `tap` hard to read relative to creating a local and returning the local. 

Hmm... what about.. (I may have the splats wrong here...)
```
def build_a_notice(criterion, *args)
  Notice.new(*args) if public_send(criterion)
end

def build_notices(*notice_args)
  notice_args.map { |criterion, args| build_a_notice(criterion, args) }.compact
end

def statuses 
  build_notices([:in_review?, :in_review], [:in_dispute?, :in_dispute], [:can_reconcile_journaled?, :can_reconcile], [:in_open_journal?, :in_open_journal])
end

def warnings 
  build_notices(:problem, problem_description_key, :warning)
end
```

The build methods could be class methods of `Notice`, too...